### ISSUE TITLE
Make sourcemap upload to fail in cases we know in advance they won't be applied

### ISSUE THREAD

**Issue by jalvz** [elastic/apm-server#1128]
Basically cover the unhappy path for the cases in which output configuration is not compatible with sourcemap processing, if any.

As per this comment: https://github.com/elastic/apm-server/pull/1114#discussion_r201928536

**Comment by graphaelli** [comment]
-1 on this - I don't see why we need to artificially limit this functionality.  I think it would be reasonable for a user to set up apm-server(s) to output events to logstash for enrichment (or otherwise) before insertion to elasticsearch but still use elasticsearch directly as a sourcemap repository.  

I do concede that it's probably not wonderful to upload sourcemaps through some outputs as they tend to be large and limits could easily be hit for maximum payload size (eg `message.max.bytes` in kafka) but still don't see that as a reason to limit already functional use.

**Comment by simitt** [comment]
I probably have mis-judged the situation, as I thought the sourcemap upload also fails when it is not configured properly, but I took another look and @graphaelli is right that this is not necessarily the case. 

With the current implementation, if no ES endpoint is specified, the sourcemap upload might work (out of our control then), but it cannot be applied to documents as this requires ES configuration.

**Comment by jalvz** [comment]
This depends on 1114 which is still not merged, but if at any time there is a situation in which we tell the user `"Unable to determine sourcemap storage, sourcemaps will not be applied"`, shouldn't we tell the user also `"Unable to determine sourcemap storage, sourcemaps will not be uploaded"`?

I think this is bad experience:

```
curl -X POST .../sourcemaps -F sourcemap=@...
```
> 202 Accepted
```
curl -X POST .../errors -d @...
```
> "Unable to determine sourcemap storage"

If I were experiencing that i'd say, well, then why you told me that the upload was accepted?

**Comment by graphaelli** [comment]
Following discussion, we'll make the original `202 Accepted` mention that sourcemaps will not be applied if no elasticsearch output is configured.

**Comment by jalvz** [comment]
> since we don't normally return a body with 202s I expect no one will ever see it
Fair point, opened this one then: https://github.com/elastic/apm-server/issues/1140

At very least, Apm Server can log it as a warning? 
I still think current situation is bad user experience
